interface/d3.network.js

```diff
@@ -0,0 +1,679 @@
+        if (!drawn && geneColorEnabled) {
+            addGeneLegend();
+        }
+        else if (!geneColorEnabled && drawn) {
+            d3.select("svg.gene-legend").remove();
+        }
+    };
+
+    my.showEdgeLegend = function(x) {
+        if (!arguments.length) {
+          x = true;
+        }
+        edgeLegendEnabled = x;
+        return my;
+    };
+
+    function set_draw_genes() {
+        // Remove hanging nodes: calculate node degrees to the filtered edges
+        var n = genes.length,
+            m = draw_edges.length,
+            edge, i;
+        for( i = 0; i < n; i++ ) {
+            genes[i].draw_degree = 0;
+        }
+        for( i = 0; i < m; i++ ) {
+            edge = draw_edges[i];
+            edge.source.draw_degree += Math.abs(edge.weight);
+            edge.target.draw_degree += Math.abs(edge.weight);
+        }
+        draw_genes = draw_genes.filter(function(d) {
+            return d.draw_degree > 0 || d.query; })
+    }
+
+    my.filter = function(edge_cut, node_cut) {
+        var gene_filter = function(d) {
+            return d.query || d.rank < node_cut;
+        };
+
+        var edge_filter = function(d) {
+            return d.weight > edge_cut && gene_filter(d.target)
+            && gene_filter(d.source);
+        };
+
+        draw_genes = genes.filter(gene_filter);
+        draw_edges = edges.filter(edge_filter);
+        set_draw_genes();
+
+        return my;
+    };
+
+    my.filterWithWeightSign = function(min_edge_cut, max_edge_cut, node_cut,
+                                       weightSign) {
+        var gene_filter = function(d) {
+            return d.query || d.rank < node_cut;
+        };
+
+        var edge_filter = function(d) {
+            if (max_edge_cut < 0) { // Input edge_cut values must be positive
+              return false;
+            }
+
+            if (!gene_filter(d.target) || !gene_filter(d.source)) {
+              return false;
+            }
+
+            var cmpVal;
+            if (!weightSign) {
+                cmpVal = Math.abs(d.weight);
+            } else if (weightSign < 0) {
+                cmpVal = -d.weight;
+            } else if (weightSign > 0) {
+                cmpVal = d.weight;
+            } else {  // in case cmpVal is not even a valid number
+                cmpVal = Math.abs(d.weight);
+            }
+
+            return cmpVal > min_edge_cut && cmpVal < max_edge_cut;
+        };
+
+        draw_genes = genes.filter(gene_filter);
+        draw_edges = edges.filter(edge_filter);
+        set_draw_genes();
+
+        return my;
+    };
+
+    my.onEdge = function(type, listener) {
+        if ( typeof listener == "function" ) edge_events[type] = listener;
+        return my;
+    };
+
+    my.edgeColor = function(x) {
+        if (!arguments.length) return edgeColor;
+        if ( typeof x == "function" ) edgeColor = x;
+        else edgeColor = function(d) { return x; };
```

Comment:
No, this is the trick that has been used very often in this library. It means if `edgeColor()` is invoked like this:
```network.edgeColor("red");```
then it resets the internal `edgeColor` variable into such a function:
```
edgeColor = function(d) {
    return "red";
}
```
Technically speaking, the parameter `d` is unnecessary, because it is not being used. But since it was written by Aaron, I never bother.

Reply:
Okay, gotcha. This looked funky at first, but when I started seeing it lots of places I started to realize it must have been intentional.